platforms/sdl/main.cpp

```diff
@@ -131,46 +146,24 @@ static void handle_events()
 	}
 }
 
-// GUI Scale
-static void calculate_gui_scale()
-{
-	int width = Minecraft::width;
-
-	// Modified Version Of https://github.com/MCPI-Revival/Ninecraft/blob/3f71638a10b581f6a50669edb24bc1ef1a92fbea/ninecraft/src/main.c#L243-L255
-	if (width < 1000)
-	{
-		if (width < 400)
-		{
-			Gui::InvGuiScale = 1.0f;
-		}
-		else
-		{
-			Gui::InvGuiScale = 0.5f;
-		}
-	}
-	else
-	{
-		Gui::InvGuiScale = 0.25f;
-	}
-}
-
 // Resizing
 static void resize()
 {
-	SDL_GL_GetDrawableSize(window, &Minecraft::width, &Minecraft::height);
-
-	if (g_pApp != nullptr)
-	{
+    int drawWidth, drawHeight;
```

Comment:
The drawWidth, drawHeight, windowWidth and windowHeight ought to match. Otherwise, it will look blurry on retina displays on MacOS. Maybe increase the size of the GUI artificially if the screen size is over a certain width, like 2800 pixels? (4k screens have this problem as well)

Reply:
What you're describing is the exact opposite of how it actually works. The draw & window dimensions provided by SDL do not match on retina displays. The changes I've made have fixed all of the scaling issues found on retina displays. This has been tested on a 2015 MBP.

Reply:
Interesting behavior. I guess shouldn't you then use the drawWidth and drawHeight for everything? (since the opengl surface is drawWidth x drawHeight in size)

Reply:
Oh, wait, does SDL seriously send events divided by the draw scale? That's so messed up, yeah, probably makes more sense now that I look into it better.

Reply:
The OS sends events through SDL in a point-based system rather than by pixel. This is only made apparent when high DPI is used and causes a desync between the resolution and the DPI.

Reply:
Alright, this is cleared up then.